dcos/dcos-ui
# [spike] DCOS-40571: Filter by Type on the Nodes Agents view (both on list and grid view)

Add the option to filter the nodes by
public and private for better user experience.

Closes https://jira.mesosphere.com/browse/DCOS-40571

## Testing
1. Go to Nodes tab.
2. Open the filter dropdown.
3. Verify that the type filter is there and it works.
4. Verify that typing `is:private` and `is:public` in the search field also works.
5. Verify that the added tests make sense.

## Trade-offs
I know that we want to have all new code in typescript, but I don't think it makes sense to do it when copy-pasting. I think a better solution for the future is to abstract these sections and filters.

## Dependencies
None.

## Screenshots
![Peek 2019-05-23 15-40](https://user-images.githubusercontent.com/40791275/58253532-586fa900-7d71-11e9-9374-29508778761a.gif)
